presentation/src/main/java/com/wednesday/template/presentation/weather/home/HomeFragment.kt

```diff
@@ -37,11 +39,18 @@ class HomeFragment : MainFragment<
         }
     }
 
+    private val snackbarComponent by component {
+        SnackbarComponent(this)
+    }
+
     override fun onState(screenState: HomeScreenState) {
         super.onState(screenState)
         listComponent.setData(screenState.items)
     }
 
     override fun onEffect(effect: Effect) {
+        when (effect) {
+            is ShowSnackbarEffect -> snackbarComponent.setData(effect)
+        }
```

Comment:
We should also add an else branch to make this exhaustive. This will become an error in the future kotlin versions.
On the else branch call `unhandledEffect(effect)` so that we throw an exception and don't forget to handle any effects that are set from the view model.